src/data_structure/binary_tree.rs

```diff
@@ -0,0 +1,66 @@
+#[derive(PartialEq)]
+pub struct Node<'a> {
+    val: &'a str,
+    left: Option<Box<Node<'a>>>,
+    right: Option<Box<Node<'a>>>,
+}
+
+impl<'a> Node<'a> {
```

Comment:
I was thinking of having different types of nodes, like for a Binary Tree, for B-Tree.

And then different traits for different kinds of trees.
Like for `BinaryTree`, the trait will have an `insert` method, which is simple, just insert wherever you want.
But for `BinarySearchTree` the trait's method will impose this condition of `Order` between among children.